opencart 1.5.6.4/upload/admin/controller/module/smaily_for_opencart.php

```diff
@@ -0,0 +1,333 @@
+ * @license GPL-3.0+
+ * @copyright 2019 Smaily
+ *
+ * Plugin Name: Smaily for Opencart
+ * Description: Smaily email marketing and automation extension plugin for Opencart.
+ * Version: 1.1.3
+ * License: GPL3
+ * Author: Smaily
+ * Author URI: https://smaily.com/
+ *
+ * Smaily for Opencart is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * any later version.
+ *
+ * Smaily for Opencart is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Smaily for Opencart. If not, see <http://www.gnu.org/licenses/>.
+ */
+class ControllerModuleSmailyForOpencart extends Controller {
+    private $error = array();
+
+    public function index() {
+        // Add language file.
+        $this->load->language('module/smaily_for_opencart');
+        // Settings model.
+        $this->load->model('setting/setting');
+        // Smaily admin page model.
+        $this->load->model('smailyforopencart/admin');
+        // Add heading title.
+        $this->document->setTitle($this->language->get('heading_title'));
+        
+        // When save is pressed.
+        if (($this->request->server['REQUEST_METHOD'] == 'POST') && ($this->validate())) {
+
+            // Get validate field from db.
+            $validated = $this->model_smailyforopencart_admin->getSettingValue('smaily_for_opencart_validated');
+
+            // Read form data.
+            $data = $this->request->post;
+            // Append validated status to form data.
+            if ($validated) {
+                $data['smaily_for_opencart_validated'] = $validated;
+            }
+            // Read validated credentials from database
+            $data['smaily_for_opencart_subdomain'] = $this->model_smailyforopencart_admin->getSettingValue('smaily_for_opencart_subdomain');
+            $data['smaily_for_opencart_username'] = $this->model_smailyforopencart_admin->getSettingValue('smaily_for_opencart_username');
+            $data['smaily_for_opencart_password'] = $this->model_smailyforopencart_admin->getSettingValue('smaily_for_opencart_password');
+            // Save settings
+            $this->model_setting_setting->editSetting('smaily_for_opencart', $data);
+            // Display Success after pressing save
+            $this->session->data['success'] = $this->language->get('text_success');
+            // Redirect to module settings page.
+            $this->redirect($this->url->link('extension/module', 'token=' . $this->session->data['token'], 'SSL'));
+        }
+
+        // Text fields
+        $this->data['heading_title'] = $this->language->get('heading_title');
```

Comment:
Clean up variables. For example `button_save` and `button_cancel` are declared, but not used by the template.